includes/shortcodes/class-wc-shortcode-my-account.php

```diff
@@ -39,7 +39,7 @@ public static function output( $atts ) {
 			return;
 		}
 
-		if ( ! is_user_logged_in() ) {
+		if ( ! is_user_logged_in() || isset( $wp->query_vars['lost-password'] ) ) {
```

Comment:
I think we should add another check, perhaps inline 54 or inside the lost password method, to make sure that if the user is logged in, then it's the same user who is trying to reset the password. This is to make it harder for any external plugin to accidentally switch users in the password reset action in case a user clicks on a password reset link while logged in as a different user. wdyt?

Reply:
That would be extra safe, I like this idea. Will take a look at that next :)

Reply:
Good catch! I tested this and it was possible to set another user's password. (Yikes!)

I tried two different fixes for this (see commit history if you're curious). 

The fix I went with aborts the password reset as early as possible, before redirecting to the set-password form 5e3713da75b68fa46874309b56f2947ab93134e0. I've also added a notice for the rare case that a user tries to use a password reset link in a browser session logged in as a different user.

Thanks for catching this @vedanshujain. Can you give it another review / test to be sure? Thanks again 🙇

Reply:
By the way, if you think it's necessary, we can add the check in `lost_password` as well, then the code is even more robust.

Reply:
I've added more info to the description about this change + a screenshot of the notice for clarity.